client/CMakeLists.txt

```diff
@@ -26,6 +26,7 @@ INCLUDE_DIRECTORIES(
   ${CMAKE_SOURCE_DIR}/strings
   ${MY_READLINE_INCLUDE_DIR}
   ${CMAKE_CURRENT_BINARY_DIR}
+  SYSTEM ${BOOST_PATCHES_DIR}
```

Comment:
If my understanding of CMake docs is correct, I don't think it works like this. "SYSTEM" has to come first thing in INCLUDE_DIRECTORIES and will apply to the whole directive, which here has to be split in two: one SYSTEM and one regular

Reply:
Fixed.